pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -1320,14 +1390,34 @@ func getSHCPlaybookContext(ctx context.Context, client splcommon.ControllerClien
 }
 
 // getLocalScopePlaybookContext returns the local scoped app install playbook context
-func getLocalScopePlaybookContext(ctx context.Context, installWorker *PipelineWorker, sem chan struct{}, podExecClient splutil.PodExecClientImpl) PlaybookImpl {
+func getLocalScopePlaybookContext(ctx context.Context, installWorker *PipelineWorker, sem chan struct{}, podExecClient splutil.PodExecClientImpl) *localScopePlaybookContext {
 	return &localScopePlaybookContext{
 		worker:        installWorker,
 		sem:           sem,
 		podExecClient: podExecClient,
 	}
 }
 
+// getInsallWorkerPlaybookContext returns the playbook context for install workers i.e either local
+// or premiumApps scope for now
+func getInsallWorkerPlaybookContext(ctx context.Context, worker *PipelineWorker, sem chan struct{}, podExecClient splutil.PodExecClientImpl, appSrcSpec *enterpriseApi.AppSourceSpec, appSrcScope string, ppln *AppInstallPipeline) PlaybookImpl {
+	reqLogger := log.FromContext(ctx)
+	scopedLog := reqLogger.WithName("getInsallWorkerPlaybookContext").WithValues("crName", ppln.cr.GetName(), "namespace", ppln.cr.GetNamespace())
+
+	// Since local app context is needed for premiumAppContext we retrieve it for both cases
+	localCtx := getLocalScopePlaybookContext(ctx, worker, sem, podExecClient)
+	if appSrcScope == enterpriseApi.ScopeLocal {
+		return localCtx
+	} else if appSrcScope == enterpriseApi.ScopePremiumApps {
+		return getPremiumAppScopePlaybookContext(ctx, localCtx, appSrcSpec, ppln.client, ppln.cr, ppln)
+	}
```

Comment:
I think the better way is this - 
```
        // Since local app context is needed for premiumAppContext we retrieve it for both cases
	if appSrcScope == enterpriseApi.ScopeLocal {
		return getLocalScopePlaybookContext(ctx, worker, sem, podExecClient)
	} else if appSrcScope == enterpriseApi.ScopePremiumApps {
		return getPremiumAppScopePlaybookContext(ctx, localCtx, appSrcSpec, ppln.client, ppln.cr, ppln)
	}

Reply:
We are using the variable `localCtx` of type `localScopePlaybookContext` as an argument to the `getPremiumAppScopePlaybookContext` API. Hence we need to compute the localCtx first for both cases.

Reply:
oh okay, completely missed it :)